base/linalg/triangular.jl

```diff
@@ -1888,10 +1898,7 @@ function sqrtm{T}(A::UpperTriangular{T})
     for j = 1:n
```

Comment:
It seems like we should just have `sqrtm(A)` call a `sqrtm{realmatrix}(A,::Val{realmatrix})` function with either `Val{true}` or `Val{false}` so that every operation on `R` is type-stable, and you don't need `floop`.